### PR TITLE
Make `ilab` able to download safetensors

### PR DESCRIPTION
**Pull request by cdoern**
when downloading from HF, check check if the repo has .safetensor files. If it does, and `--filename` is default, download the entire repo and put it into a <repo_name> folder inside of models

Signed-off-by: Charlie Doern <cdoern@redhat.com>

### CODE REVIEW

File: src/instructlab/lab.py
Comment (tiran) [inline src/instructlab/lab.py:729]:
```suggestion
            os.makedirs(os.path.join(model_dir, folder_name), exist_ok=True)
```

File: src/instructlab/lab.py
Comment (oindrillac) [inline src/instructlab/lab.py:720]: Should we also update the readme user docs which point to the additional steps (this and stuff like model-family etc)? They can be a separate PR in the interest of time. For now maybe we can make the error message more descriptive

```suggestion
            "HF_TOKEN var needs to be set in your environment to download HF Model. The HF Token is used to authenticate your identity to the Hugginface Hub."
```

Comment (alinaryan) [reply src/instructlab/lab.py:720]: small nit in case you commit this: 
's/Hugginface/Huggingface/'

**Review by russellb — APPROVED** [review]
lgtm!  I know @alimaredia is testing this, so I requested his review. It should block merging until he approves as well.

File: src/instructlab/lab.py
Comment (jaideepr97) [inline src/instructlab/lab.py:717]: do we need this check? I think it might be best not to hardcode merlinite for such checks - it would also be good practice to always have an HF token exposed and would make the code easier and cleaner to maintain (even if technically you dont need a token for the instructlab org)

just me 0.02

Comment (russellb) [reply src/instructlab/lab.py:717]: I assume this was so at least the default config didn't require it, which is nice.

An even better UX would determine whether the token is required for a given model before then erroring out because it's not set, but that could be a future improvement ...

Comment (jaideepr97) [reply src/instructlab/lab.py:717]: right - then for now should we at least only check if "instructlab" is part of the repo string, and we allow the download without token if true? since we would want to at least have uniform behavior for all the models in the `instructlab` HF org

Comment (tiran) [reply src/instructlab/lab.py:717]: How about you use clicks' feature to read [values from env vars](https://click.palletsprojects.com/en/8.1.x/options/#values-from-environment-variables)?
`@click.option('--hf-token', envvar='HF_TOKEN')`

Comment (cdoern) [reply src/instructlab/lab.py:717]: merl is the default. This is my way of checking `If NO HF TOKEN and THE USER DIDN'T SET --FILENAME`

File: src/instructlab/lab.py
Comment (alimaredia) [inline src/instructlab/lab.py:725]: @cdoern why not download the entire repository via `snapshot_download()` instead of going file by file? https://huggingface.co/docs/huggingface_hub/en/guides/download#download-an-entire-repository

File: src/instructlab/lab.py
Comment (russellb) [reply src/instructlab/lab.py:717]: @tiran yes! I was just looking at that yesterday when looking at https://github.com/instructlab/instructlab/pull/1022

File: src/instructlab/lab.py
Comment (russellb) [reply src/instructlab/lab.py:725]: This looks like a good idea to me +1

Comment (russellb) [reply src/instructlab/lab.py:725]: Apologies @alimaredia -- once you posted this review, that made mergify happy that you looked at it.  If you had done "request changes" it would have blocked it, just FYI.

I probably should just set a `hold` label and let you remove it

**Comment by cdoern** [comment]
uh oh, seems mergify has gone rogue!!!! I will open another PR to address folks' comments

File: src/instructlab/lab.py
Comment (bbrowning) [reply src/instructlab/lab.py:717]: I was surprised by it asking me for a HF_TOKEN when I did `ilab download --repository instructlab/merlinite-7b-lab` as that repository does not need one. In my case, I specifically wanted it to download the non-gguf version of this model, but perhaps the logic around HF_TOKEN usage should be a bit broader?

Comment (russellb) [reply src/instructlab/lab.py:717]: This was improved further in https://github.com/instructlab/instructlab/pull/1035

**Pull request merged**
